backend/danswer/bots/slack/handlers/handle_message.py

```diff
@@ -178,15 +179,23 @@ def _get_answer(question: QuestionRequest) -> QAResponse:
 
     answer_failed = False
     try:
+        # By leaving time_cutoff and favor_recent as done, and setting enable_auto_detect_filters
+        # it allows the slack flow to extract out filters from the user query
+        filters = RequestFilters(
+            source_type=None,
+            document_set=document_set_names,
+            time_cutoff=None,
+        )
+
         # This includes throwing out answer via reflexion
         answer = _get_answer(
             QuestionRequest(
                 query=msg,
                 collection=DOCUMENT_INDEX_NAME,
                 use_keyword=False,  # always use semantic search when handling Slack messages
-                filters=[{DOCUMENT_SETS: document_set_names}]
-                if document_set_names
-                else None,
+                enable_auto_detect_filters=not DISABLE_DANSWER_BOT_FILTER_DETECT,
```

Comment:
nit: make this an arg to the function with default value like other configs

Reply:
XD boooo ok